airbyte-cdk/bulk/core/load/src/testFixtures/kotlin/io/airbyte/cdk/load/write/BasicFunctionalityIntegrationTest.kt

```diff
@@ -573,6 +574,166 @@ abstract class BasicFunctionalityIntegrationTest(
         )
     }
 
+    /**
+     * Test behavior in a failed truncate refresh. Sync 1 just populates two records with ID 1 and
+     * 2. The test then runs two more syncs:
+     * 1. Sync 2 emits ID 1, and then fails the sync (i.e. no COMPLETE stream status). We expect the
+     * first sync's records to still exist in the destination, and the new records to not be visible
+     * to the data dumper yet. This miiight not be correct - we should doublecheck this when we
+     * implement bigquery. Warehouse destinations might commit data for non-truncate syncs on
+     * failure and I don't want to track down the official refreshes spec for this right now
```

Comment:
.... actually, what do we do in s3v2 if staging is disabled? I guess we still defer the deletion until we get a success status, but we write directly to the output directory?

Reply:
Yes, that's the current behavior

Reply:
Yeah. The difference is that after failure:
* if staging enabled: partially synced data will still be in staging
* otherwise: partially synced data will already be in final

Reply:
Do we want a flag for the non staging behavior, like "commitsIncrementally"?

Reply:
makes sense, I'll add that flag

Reply:
added the flag, it looks like s3v2 isn't correctly retaining records from the 2nd sync. Maybe will be fixed by your destination state handling stuff? I'm going to leave the test disabled for now in s3 and merge this PR as-is